components/Switch.tsx

```diff
@@ -0,0 +1,401 @@
+import { Input } from "./ui/input";
+import { Button } from "./ui/button";
+import { BusyDialog } from "./BusyDialog";
+import { SendErrorDialog } from "./SendErrorDialog";
+import { SubmittableExtrinsic } from "@polkadot/api/types";
+import { ISubmittableResult } from "@polkadot/types/types";
+import PolkadotBalance from "./Balances";
+import { parseUnits } from "ethers";
+import { toast } from "sonner";
+import { parachainConfigs } from "@/utils/parachainConfigs";
+
+import { useRouter } from "next/navigation";
+import { TopUpXcmFee } from "./TopUpXcmFee";
+import { toPolkadot } from "@snowbridge/api";
+import { LocationSelector } from "./LocationSelector";
+
+export const SwitchComponent: FC = () => {
+  const snowbridgeEnvironment = useAtomValue(snowbridgeEnvironmentAtom);
+  const context = useAtomValue(snowbridgeContextAtom);
+  const polkadotAccounts = useAtomValue(polkadotAccountsAtom);
+  const polkadotAccount = useAtomValue(polkadotAccountAtom);
+  const router = useRouter();
+
+  const [feeDisplay, setFeeDisplay] = useState("");
+  const [error, setError] = useState<ErrorInfo | null>(null);
+  const [busyMessage, setBusyMessage] = useState("");
+  const [sufficientTokenAvailable, setSufficientTokenAvailable] =
+    useState(true);
+  const [topUpCheck, setTopUpCheck] = useState(false);
+  const [transaction, setTransaction] = useState<SubmittableExtrinsic<
+    "promise",
+    ISubmittableResult
+  > | null>(null);
+  const [tokenSymbol, setTokenSymbol] = useState<string | null>(null);
+
+  const filteredLocations = useMemo(
+    () =>
+      snowbridgeEnvironment.locations
+        .filter((x) => x.type !== "ethereum")
+        .filter((x) => x.name !== "Muse"),
+    [snowbridgeEnvironment],
+  );
+
+  const initialDestination = useMemo(
+    () => filteredLocations.find((v) => v.id === "KILT"),
+    [filteredLocations],
+  );
+
+  const form: UseFormReturn<FormDataSwitch> = useForm<
+    z.infer<typeof formSchemaSwitch>
+  >({
+    resolver: zodResolver(formSchemaSwitch),
+    defaultValues: {
+      source: filteredLocations.find((v) => v.id === "assethub"),
+      // to do: handle this better
+      destination: initialDestination ?? "",
+      token: initialDestination?.erc20tokensReceivable[0].id,
+      beneficiary: polkadotAccount?.address ?? "",
+      sourceAccount: polkadotAccount?.address ?? "",
+      amount: "0.0",
+    },
+  });
+
+  const source = form.watch("source");
+  const destination = form.watch("destination");
+  const sourceAccount = form.watch("sourceAccount");
+  const beneficiary = form.watch("beneficiary");
+  const amount = form.watch("amount");
+  const token = form.watch("token");
+
+  const beneficiaries: AccountInfo[] = useMemo(
+    () =>
+      polkadotAccounts?.map((x) => ({
+        key: x.address,
+        name: x.name || "",
+        type: destination.type,
+      })) || [],
+    [polkadotAccounts, destination.type],
+  );
+
+  const handleTransaction = useCallback(async () => {
```

Comment:
the name is misleading, it's `createTransaction` isn't it?

Reply:
https://github.com/Snowfork/snowbridge-app/pull/23/commits/7a77887364dd2c60c50a3ec3a301cd0939e49663